archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,97 @@
+import java.nio.file.Files
+import org.apache.commons.io.FileUtils
+
+def jakartaVersion = request.properties["jakartaVersion"].trim()
+def docker = request.properties["docker"].trim()
+def outputDirectory = new File(request.getOutputDirectory(), request.getArtifactId())
+
+validateInput(jakartaVersion, outputDirectory);
+
+switch (request.properties["runtime"]) {
+    case "glassfish": println "Generating code for GlassFish"
+        if (docker.equalsIgnoreCase("yes")) {
+           println "WARNING: GlassFish does not support Docker"
+           FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))           
+        }
+        
+        break
+
+    case "tomee": println "Generating code for TomEE"
+        break
+
+    case "payara": println "Generating code for Payara"
+        if ((jakartaVersion != '8') && docker.equalsIgnoreCase("yes")) {
+           println "WARNING: Payara does not yet support Docker for this Jakarta EE version"
+           FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))
+        }
+            
+        break
+
+    default: println "No runtime will be included in the sample"
+}
+
+// Jakarta version specific processing
+bindEEPackage(jakartaVersion, outputDirectory)
+
+// Remove Dockerfile if not requested or possible
+if (docker.equalsIgnoreCase("no")) {
+    println "Docker support was not requested"
+    FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))
+} else if (request.properties["runtime"].equalsIgnoreCase("none")) {
+    println "WARNING: Docker support is not possible without choosing a runtime"
+    FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))    
+}
+
+// Adding Maven Wrapper
+println "Adding Maven Wrapper"
```

Comment:
Code to generate the Maven wrapper could go to a separate method, like `generateMavenWrapper()`

Reply:
Good point. Done!